### PR TITLE
Migrate preferences to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/preferences.dart [hunk 1]
```diff
@@ -23,22 +23,22 @@ class PreferencesController {
   Future<void> init() async {
     if (storage != null) {
       // Get the current values and listen for and write back changes.
-      String value = await storage.getValue('ui.darkMode');
-      toggleDarkModeTheme(value == null || value == 'true');
+      String value = await storage!.getValue('ui.darkMode');
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/preferences.dart:26]: done

File: packages/devtools_app/lib/src/shared/globals.dart [hunk 2]
```diff
@@ -23,7 +23,7 @@ MessageBus get messageBus => globals[MessageBus];
 
 FrameworkController get frameworkController => globals[FrameworkController];
 
-Storage get storage => globals[Storage];
+Storage? get storage => globals[Storage];
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/globals.dart]: If I do this, the test test/debugger_evalution_test.dart starts failing as it does not have serviceManager set, and I do not know how to fix it.
I tried to add setGlobal(ServiceConnectionManager, FakeServiceManager()); to setUp, but the test started timing out.
Can you suggest what to do about it?

[the remark this answers is not in the document]
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/shared/globals.dart]: It looks like the first failure is related to Storage being null, not service manager. 

```
01:08 +115 -1: /home/runner/work/devtools/devtools/packages/devtools_app/test/debugger_evalution_test.dart: EvalOnDartLibrary returns scoped variables when EditingParts is not a field [E]            
  NoSuchMethodError: The method 'getValue' was called on null.
  Receiver: null
  Tried calling: getValue("ui.darkMode")
  dart:core                                               Object.noSuchMethod
  package:devtools_app/src/shared/preferences.dart 24:35  PreferencesController.init
  test/test_infra/flutter_test_environment.dart 117:35    FlutterTestEnvironment.setupEnvironment
  ```

Try adding `setGlobal(Storage, BrowserStorage());` here https://github.com/flutter/devtools/blob/master/packages/devtools_app/test/test_infra/flutter_test_environment.dart#L120.  You may have to play with a few different values for what to set the Storage global to. Search for `setGlobal(Storage` and you'll see the different storage classes we set in DevTools.

[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/globals.dart]: it worked. thank you!

**Review by kenzieschmoll — APPROVED** [review]

**Comment by polina-c** [comment]
tested

**Pull request merged**